src/zkevm_specs/state.py

```diff
@@ -375,36 +375,16 @@ def check_account_destructed(row: Row, row_prev: Row):
 def check_tx_log(row: Row, row_prev: Row):
     # tx_id | log_id | field_tag | index | value
     tx_id = row.id()
-    pre_tx_id = row_prev.id()
-    log_id = row.address()
-    pre_field_tag = row_prev.field_tag()
-    field_tag = row.field_tag()
-    index = row.storage_key()
-    pre_index = row_prev.storage_key()
+    prev_tx_id = row_prev.id()
 
     # 12.0 is_write is always true
     assert row.is_write == 1
-    # 12.1 reset log_id when tx_id increases
+    # 12.1 tx_id change must be increasing
```

Comment:
I think "tx_id change is either 0 or 1" would be more accurate.

Reply:
since state circuit sorts by `id` key , I think it is should be increasing by default, otherwise lexicographic ordering will fail,  right ?   
if so , this check can also be removed here, generally we can skip all key increasing check in all type Ops state constraints!

Reply:
It's true that the lexicographic ordering chip forces the id to be increasing here, but actually it's the evm circuit that imposes the stronger constraint that the change must be either 0 or 1.

Reply:
evm circuit has no constrain for specific log operation.   I just realize that tx id change can be any value , because some tx may don't contain log* step, so the tx_id is not continuous  in the rw table. such as tx_1, tx_2...tx_5, only tx_1 and tx_5 contain log step.  we have table like this: 

| rw_counter  | tx_id      | log_id |  ..    |
| -----     | ---- | ----- | --- |
| 1   |  1      |   1       | ..  |
| 2  |   5     |     1        | ..  |

so I think we can remove the tx_id constrain since lexicographic ordering force it to be increasing .  
@ed255  WDYT?

Reply:
removed tx id constraint in [b7d16ff](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/220/commits/b7d16fff6027d8a70ca203c92c3afe5974fe07eb)

Reply:
> evm circuit has no constrain for specific log operation. I just realize that tx id change can be any value , because some tx may don't contain log* step, so the tx_id is not continuous in the rw table. such as tx_1, tx_2...tx_5, only tx_1 and tx_5 contain log step. we have table like this:
> rw_counter 	tx_id 	log_id 	..
> 1 	1 	1 	..
> 2 	5 	1 	..
> 
> so I think we can remove the tx_id constraint since lexicographic ordering force it to be increasing . @ed255 WDYT?

I think you're right!  The constraint for `tx_id` increasing by `+1` will not always be satisfied, so it should be removed.